src/LendingPool.sol

```diff
@@ -588,25 +648,53 @@ contract LendingPool is Guardian, TrustedCreditor, DebtToken, InterestRateModule
             }
             tranche = tranches[i];
             maxBurned = realisedLiquidityOf[tranche];
-            if (assets < maxBurned) {
+            if (badDebt < maxBurned) {
                 // burn
-                realisedLiquidityOf[tranche] -= assets;
-                totalRealisedLiquidity -= assets;
+                realisedLiquidityOf[tranche] -= badDebt;
                 break;
             } else {
-                ITranche(tranche).lock();
+                ITranche(tranche).lock(); //todo gas: can be removed
                 // burn
                 realisedLiquidityOf[tranche] -= maxBurned;
-                totalRealisedLiquidity -= maxBurned;
                 _popTranche(i, tranche);
                 unchecked {
-                    assets -= maxBurned;
+                    badDebt -= maxBurned;
                 }
             }
         }
+    }
 
-        //ToDo Although it should be an impossible state if the protocol functions as it should,
-        //What if there is still more liquidity in the pool than totalRealisedLiquidity, start an emergency procedure?
+    /**
+     * @notice Syncs liquidation penalties to the Lending providers and the treasury.
+     * @param assets The total amount of underlying assets to be paid out as liquidation penalty.
+     * @dev The liquidationWeight of each Tranche determines the relative share yield (interest payments) that goes to its Liquidity providers.
+     */
+    function _syncLiquidationPenaltyToLiquidityProviders(uint256 assets) internal {
+        uint256 remainingAssets = assets;
+
+        uint256 trancheShare;
+        uint256 weightOfTranche;
+        for (uint256 i; i < tranches.length;) {
+            weightOfTranche = liquidationWeightTranches[i];
+
+            if (weightOfTranche != 0) {
+                //skip if weight is zero, which is the case for Sr tranche
+                trancheShare = uint256(assets).mulDivDown(weightOfTranche, totalLiquidationWeight);
+                unchecked {
+                    realisedLiquidityOf[tranches[i]] += trancheShare;
+                    remainingAssets -= trancheShare;
+                }
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        unchecked {
+            // Add the remainingAssets to the treasury balance
```

Comment:
an 'if zero, skip", might make sense here

Reply:
This will only be the case if liquidationWeightTreasury is 0 and we will always take a cut to the protocol no?
Think this would just be an additional unnecessary check.